handlers.go

```diff
@@ -2880,7 +2880,7 @@ func (s *Server) handleRegisterUser(c echo.Context) error {
 		}
 	}
 
-	if exist != nil {
+	if exist != nil && strings.ToLower(exist.Username) == username {
```

Comment:
@alvin-reyes why is this necessary?

Reply:
The `exists` here will always be not nil because it's pointer, e.i it will always hold the reference and even if it doesn't get any record, it will initialize a new instance of the same object.

I added a check to validate if the current record `Username` is equivalent to given `username`.

Reply:
Oh shit, I see 😞